### ISSUE TITLE
Standardise type hints

### ISSUE THREAD

**Issue by mscroggs** [UCL/causalprog#27]
We should standarise type hints for things like "array-like" and "rng seed"

**Comment by willGraham01** [comment]
I think our plan for this, for now, should be to make a `_typing` module or something, and inside it create a few `TypeVar`s or `TypeAlias`es that cover the things we need to standardise.

Then go through the package and typehint accordingly - that should hopefully avoid us having to do this again some time in the future... It also lets us helpfully avoid commitment to a particular framework for now (though `jax` should probably be our default).

EG (imports / correct class names not promised)

_typing.py
```python

RNGKey: TypeAlias = jax.Array | numpy.Generator
ArrayLike: TypeAlias = jax.Array | numpy.typing.ArrayLike
```

some_other_file.py
```python
from causalprog._typing import RNGKey


def random_function(rng: RNGKey, ...):
    ...
```
